{"cells":[{"cell_type":"markdown","metadata":{"id":"V_6GaxSxGDDL"},"source":["Project\n","\n","2.- How does SARS-Cov-02 infection alter transcripyional acitivity. Overall design: Comparative gene expression profiling analysis of RNA-seq data using 4 types (uninfected/infected, young/mid-aged) of isolated lung endothelial cells.\n","\n"," SRAID      | Age    |  Treatment |\n","------------|--------|------------|\n","SRR24206824 | Young  | SARS-CoV-2\n","SRR24206825 | MidAge | SARS-CoV-2\n","SRR24206826 | Young  | PBS\n","SRR24206827 | MidAge | PBS"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"qadKhzojD_HC"},"outputs":[],"source":["# Run this box as this! It is installing the NCBIs SRA toolkit! This is how we will be dowloading data.\n","!wget https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/3.1.0/sratoolkit.3.1.0-ubuntu64.tar.gz .\n","!tar -vxzf sratoolkit.3.1.0-ubuntu64.tar.gz\n","!chmod +777 sratoolkit.3.1.0-ubuntu64/bin/fastq-dump"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"wg-VCkzSErQQ"},"outputs":[],"source":["### Start by downloading the data; subsampling the data to 5 M reads\n","!sratoolkit.3.1.0-ubuntu64/bin/fastq-dump  -X 5000000 SRR24206824\n","!sratoolkit.3.1.0-ubuntu64/bin/fastq-dump  -X 5000000 SRR24206825\n","!sratoolkit.3.1.0-ubuntu64/bin/fastq-dump  -X 5000000 SRR24206826\n","!sratoolkit.3.1.0-ubuntu64/bin/fastq-dump  -X 5000000 SRR24206827"]},{"cell_type":"code","execution_count":2,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":19876,"status":"ok","timestamp":1713677728707,"user":{"displayName":"Emily White","userId":"02727269720942967183"},"user_tz":420},"id":"7_iIrOPGS4MO","outputId":"7746a1f2-4491-4794-ce65-2085b5c9191d"},"outputs":[{"output_type":"stream","name":"stdout","text":["Mounted at /content/drive\n"]}],"source":["from google.colab import drive\n","drive.mount('/content/drive')"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"IT_79bopcimi"},"outputs":[],"source":["!mv SRR24206824.fastq Young_SARScov2.fastq\n","!mv SRR24206825.fastq Midage_SARScov2.fastq\n","!mv SRR24206826.fastq Young_PBS.fastq\n","!mv SRR24206827.fastq Midage_PBS.fastq"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"S9h0ad7_Z_pq"},"outputs":[],"source":["!cp Young_SARScov2.fastq /content/drive/MyDrive/SARSdata\n","!cp Midage_SARScov2.fastq /content/drive/MyDrive/SARSdata\n","!cp Young_PBS.fastq /content/drive/MyDrive/SARSdata\n","!cp Midage_PBS.fastq /content/drive/MyDrive/SARSdata"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"GgIPZ0lKa3vq"},"outputs":[],"source":["# m.mouse reference genome\n","# !wget ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/data/mm10.tar.gz\n","# !tar -xzf mm10.tar.gz\n","!cp mm10.tar.gz /content/drive/MyDrive/SARSdata /content/drive/MyDrive/SARSdata/"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"bHBjZ97AXh85"},"outputs":[],"source":["# Annotation Hash Table\n","!wget https://ftp.ensembl.org/pub/release-102/gtf/mus_musculus/Mus_musculus.GRCm38.102.gtf.gz\n","!gunzip Mus_musculus.GRCm38.102.gtf.gz"]},{"cell_type":"code","source":["# !cp /content/Mus_musculus.GRCm38.102.gtf.gz /content/drive/MyDrive/SARSdata/"],"metadata":{"id":"j1Ij83Pd66zE"},"execution_count":null,"outputs":[]},{"cell_type":"code","execution_count":null,"metadata":{"id":"B7Mi1wLeewtP"},"outputs":[],"source":["# FastQC libraries/tools\n","!sudo apt-get install -y default-jre\n","!java -version\n","!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n","!unzip fastqc_v0.11.9.zip\n","!chmod +x FastQC/fastqc"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"o0ULbrlSkqHZ"},"outputs":[],"source":["#checking read quality - htmls available in drive\n","!./FastQC/fastqc --version\n","!./FastQC/fastqc /content/drive/MyDrive/Young_PBS.fastq\n","!./FastQC/fastqc /content/drive/MyDrive/SARSdata/Young_SARScov2.fastq\n","!./FastQC/fastqc /content/drive/MyDrive/SARSdata/Midage_PBS.fastq\n","!./FastQC/fastqc /content/drive/MyDrive/SARSdata/Midage_SARScov2.fastq"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"QYwD-Mp-c2uQ"},"outputs":[],"source":["#trimming the primer/adapter region with poor quality\n","!pip install cutadapt\n","# !cutadapt -h\n","!cutadapt --cut 15 -q 28 -o /content/drive/MyDrive/SARSdata/Trimmed_Young_PBS.fastq /content/drive/MyDrive/SARSdata/Young_PBS.fastq\n","!cutadapt --cut 15 -q 28 -o /content/drive/MyDrive/SARSdata/Trimmed_Young_SARScov2.fastq /content/drive/MyDrive/SARSdata/Young_SARScov2.fastq\n","!cutadapt --cut 15 -q 28 -o /content/drive/MyDrive/SARSdata/Trimmed_Midage_SARScov2.fastq /content/drive/MyDrive/SARSdata/Midage_SARScov2.fastq\n","!cutadapt --cut 15 -q 28 -o /content/drive/MyDrive/SARSdata/Trimmed_Midage_PBS.fastq /content/drive/MyDrive/SARSdata/Midage_PBS.fastq"]},{"cell_type":"code","source":["#shortening read length for better alignment\n","!cutadapt --cut 200 -o /content/drive/MyDrive/SARSdata/Final_MidagePBS.fastq /content/drive/MyDrive/SARSdata/Trimmed_Midage_PBS.fastq\n","!cutadapt --cut 200 -o /content/drive/MyDrive/SARSdata/Final_MidageSARScov2.fastq /content/drive/MyDrive/SARSdata/Trimmed_Midage_SARScov2.fastq\n","!cutadapt --cut 200 -o /content/drive/MyDrive/SARSdata/Final_YoungPBS.fastq /content/drive/MyDrive/SARSdata/Trimmed_Young_PBS.fastq\n","!cutadapt --cut 200 -o /content/drive/MyDrive/SARSdata/Final_YoungSARScov2.fastq /content/drive/MyDrive/SARSdata/Trimmed_Young_SARScov2.fastq"],"metadata":{"id":"6z6lwrC3zuiy"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# FastQC libraries/tools\n","!sudo apt-get install -y default-jre\n","!java -version\n","!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n","!unzip fastqc_v0.11.9.zip\n","!chmod +x FastQC/fastqc\n","!./FastQC/fastqc /content/drive/MyDrive/SARSdata/Final_YoungSARScov2.fastq\n"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"5Hi6WHb8JhIF","executionInfo":{"status":"ok","timestamp":1713677802580,"user_tz":420,"elapsed":53920,"user":{"displayName":"Emily White","userId":"02727269720942967183"}},"outputId":"df9d0be1-5055-40cb-afdc-1fa8c301d76b"},"execution_count":3,"outputs":[{"output_type":"stream","name":"stdout","text":["Reading package lists... Done\n","Building dependency tree... Done\n","Reading state information... Done\n","The following additional packages will be installed:\n","  default-jre-headless fonts-dejavu-core fonts-dejavu-extra\n","  libatk-wrapper-java libatk-wrapper-java-jni libfontenc1 libxkbfile1 libxtst6\n","  libxxf86dga1 openjdk-11-jre x11-utils\n","Suggested packages:\n","  mesa-utils\n","The following NEW packages will be installed:\n","  default-jre default-jre-headless fonts-dejavu-core fonts-dejavu-extra\n","  libatk-wrapper-java libatk-wrapper-java-jni libfontenc1 libxkbfile1 libxtst6\n","  libxxf86dga1 openjdk-11-jre x11-utils\n","0 upgraded, 12 newly installed, 0 to remove and 45 not upgraded.\n","Need to get 3,720 kB of archives.\n","After this operation, 12.7 MB of additional disk space will be used.\n","Get:1 http://archive.ubuntu.com/ubuntu jammy/main amd64 default-jre-headless amd64 2:1.11-72build2 [3,042 B]\n","Get:2 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxtst6 amd64 2:1.2.3-1build4 [13.4 kB]\n","Get:3 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 openjdk-11-jre amd64 11.0.22+7-0ubuntu2~22.04.1 [214 kB]\n","Get:4 http://archive.ubuntu.com/ubuntu jammy/main amd64 default-jre amd64 2:1.11-72build2 [896 B]\n","Get:5 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-core all 2.37-2build1 [1,041 kB]\n","Get:6 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-extra all 2.37-2build1 [2,041 kB]\n","Get:7 http://archive.ubuntu.com/ubuntu jammy/main amd64 libfontenc1 amd64 1:1.1.4-1build3 [14.7 kB]\n","Get:8 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxkbfile1 amd64 1:1.1.0-1build3 [71.8 kB]\n","Get:9 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxxf86dga1 amd64 2:1.1.5-0ubuntu3 [12.6 kB]\n","Get:10 http://archive.ubuntu.com/ubuntu jammy/main amd64 x11-utils amd64 7.7+5build2 [206 kB]\n","Get:11 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java all 0.38.0-5build1 [53.1 kB]\n","Get:12 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java-jni amd64 0.38.0-5build1 [49.0 kB]\n","Fetched 3,720 kB in 1s (5,109 kB/s)\n","debconf: unable to initialize frontend: Dialog\n","debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 78, <> line 12.)\n","debconf: falling back to frontend: Readline\n","debconf: unable to initialize frontend: Readline\n","debconf: (This frontend requires a controlling tty.)\n","debconf: falling back to frontend: Teletype\n","dpkg-preconfigure: unable to re-open stdin: \n","Selecting previously unselected package default-jre-headless.\n","(Reading database ... 131015 files and directories currently installed.)\n","Preparing to unpack .../00-default-jre-headless_2%3a1.11-72build2_amd64.deb ...\n","Unpacking default-jre-headless (2:1.11-72build2) ...\n","Selecting previously unselected package libxtst6:amd64.\n","Preparing to unpack .../01-libxtst6_2%3a1.2.3-1build4_amd64.deb ...\n","Unpacking libxtst6:amd64 (2:1.2.3-1build4) ...\n","Selecting previously unselected package openjdk-11-jre:amd64.\n","Preparing to unpack .../02-openjdk-11-jre_11.0.22+7-0ubuntu2~22.04.1_amd64.deb ...\n","Unpacking openjdk-11-jre:amd64 (11.0.22+7-0ubuntu2~22.04.1) ...\n","Selecting previously unselected package default-jre.\n","Preparing to unpack .../03-default-jre_2%3a1.11-72build2_amd64.deb ...\n","Unpacking default-jre (2:1.11-72build2) ...\n","Selecting previously unselected package fonts-dejavu-core.\n","Preparing to unpack .../04-fonts-dejavu-core_2.37-2build1_all.deb ...\n","Unpacking fonts-dejavu-core (2.37-2build1) ...\n","Selecting previously unselected package fonts-dejavu-extra.\n","Preparing to unpack .../05-fonts-dejavu-extra_2.37-2build1_all.deb ...\n","Unpacking fonts-dejavu-extra (2.37-2build1) ...\n","Selecting previously unselected package libfontenc1:amd64.\n","Preparing to unpack .../06-libfontenc1_1%3a1.1.4-1build3_amd64.deb ...\n","Unpacking libfontenc1:amd64 (1:1.1.4-1build3) ...\n","Selecting previously unselected package libxkbfile1:amd64.\n","Preparing to unpack .../07-libxkbfile1_1%3a1.1.0-1build3_amd64.deb ...\n","Unpacking libxkbfile1:amd64 (1:1.1.0-1build3) ...\n","Selecting previously unselected package libxxf86dga1:amd64.\n","Preparing to unpack .../08-libxxf86dga1_2%3a1.1.5-0ubuntu3_amd64.deb ...\n","Unpacking libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n","Selecting previously unselected package x11-utils.\n","Preparing to unpack .../09-x11-utils_7.7+5build2_amd64.deb ...\n","Unpacking x11-utils (7.7+5build2) ...\n","Selecting previously unselected package libatk-wrapper-java.\n","Preparing to unpack .../10-libatk-wrapper-java_0.38.0-5build1_all.deb ...\n","Unpacking libatk-wrapper-java (0.38.0-5build1) ...\n","Selecting previously unselected package libatk-wrapper-java-jni:amd64.\n","Preparing to unpack .../11-libatk-wrapper-java-jni_0.38.0-5build1_amd64.deb ...\n","Unpacking libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n","Setting up default-jre-headless (2:1.11-72build2) ...\n","Setting up libxtst6:amd64 (2:1.2.3-1build4) ...\n","Setting up libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n","Setting up openjdk-11-jre:amd64 (11.0.22+7-0ubuntu2~22.04.1) ...\n","Setting up default-jre (2:1.11-72build2) ...\n","Setting up libfontenc1:amd64 (1:1.1.4-1build3) ...\n","Setting up fonts-dejavu-core (2.37-2build1) ...\n","Setting up fonts-dejavu-extra (2.37-2build1) ...\n","Setting up libxkbfile1:amd64 (1:1.1.0-1build3) ...\n","Setting up x11-utils (7.7+5build2) ...\n","Setting up libatk-wrapper-java (0.38.0-5build1) ...\n","Setting up libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n","Processing triggers for hicolor-icon-theme (0.17-2) ...\n","Processing triggers for libc-bin (2.35-0ubuntu3.4) ...\n","/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n","\n","Processing triggers for man-db (2.10.2-1) ...\n","Processing triggers for fontconfig (2.13.1-4.2ubuntu5) ...\n","openjdk version \"11.0.22\" 2024-01-16\n","OpenJDK Runtime Environment (build 11.0.22+7-post-Ubuntu-0ubuntu222.04.1)\n","OpenJDK 64-Bit Server VM (build 11.0.22+7-post-Ubuntu-0ubuntu222.04.1, mixed mode, sharing)\n","--2024-04-21 05:35:56--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n","Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n","Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 10249221 (9.8M) [application/zip]\n","Saving to: ‘fastqc_v0.11.9.zip’\n","\n","fastqc_v0.11.9.zip  100%[===================>]   9.77M  8.16MB/s    in 1.2s    \n","\n","2024-04-21 05:35:58 (8.16 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n","\n","Archive:  fastqc_v0.11.9.zip\n","  inflating: FastQC/cisd-jhdf5.jar   \n","   creating: FastQC/Configuration/\n","  inflating: FastQC/Configuration/adapter_list.txt  \n","  inflating: FastQC/Configuration/contaminant_list.txt  \n","  inflating: FastQC/Configuration/limits.txt  \n","  inflating: FastQC/fastqc           \n","  inflating: FastQC/fastqc_icon.ico  \n","   creating: FastQC/Help/\n","   creating: FastQC/Help/1 Introduction/\n","   creating: FastQC/Help/1 Introduction/.svn/\n","  inflating: FastQC/Help/1 Introduction/.svn/entries  \n","   creating: FastQC/Help/1 Introduction/.svn/props/\n","   creating: FastQC/Help/1 Introduction/.svn/text-base/\n","  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n","   creating: FastQC/Help/1 Introduction/.svn/tmp/\n","   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n","  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n","   creating: FastQC/Help/2 Basic Operations/\n","   creating: FastQC/Help/2 Basic Operations/.svn/\n","  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n","   creating: FastQC/Help/2 Basic Operations/.svn/props/\n","   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n","  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n","  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n","  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n","   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n","   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n","  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n","  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n","  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n","   creating: FastQC/Help/3 Analysis Modules/\n","   creating: FastQC/Help/3 Analysis Modules/.svn/\n","  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n","   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n"," extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n"," extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n"," extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n"," extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n"," extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n"," extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n"," extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n"," extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n"," extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n","   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n","   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n","  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n","   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n","   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n","  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n","  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n","  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n","  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n","  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n","  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n","  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n","  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n","  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n","  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n","  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n","  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n","  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n","  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n","  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n","  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n","  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n","  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n","  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n","  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n","  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n","  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n","  inflating: FastQC/INSTALL.txt      \n","  inflating: FastQC/jbzip2-0.9.jar   \n","  inflating: FastQC/LICENSE          \n","  inflating: FastQC/LICENSE.txt      \n","  inflating: FastQC/LICENSE_JHDF5.txt  \n","   creating: FastQC/net/\n","   creating: FastQC/net/sourceforge/\n","   creating: FastQC/net/sourceforge/iharder/\n","   creating: FastQC/net/sourceforge/iharder/base64/\n","  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n","  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n","  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n","  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n","   creating: FastQC/org/\n","   creating: FastQC/org/apache/\n","   creating: FastQC/org/apache/commons/\n","   creating: FastQC/org/apache/commons/math3/\n","   creating: FastQC/org/apache/commons/math3/analysis/\n","   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n","  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n","  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n","  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n","  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n","  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n","  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n","  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n","  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n","  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n","   creating: FastQC/org/apache/commons/math3/distribution/\n","  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n","  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n","   creating: FastQC/org/apache/commons/math3/exception/\n","  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n","   creating: FastQC/org/apache/commons/math3/exception/util/\n","  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n","  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n","   creating: FastQC/org/apache/commons/math3/random/\n","  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n","  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n","  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n","  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n","  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n","  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n","   creating: FastQC/org/apache/commons/math3/special/\n","  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n","  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n","  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n","  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n","  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n","   creating: FastQC/org/apache/commons/math3/util/\n","  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n","  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n","  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n","  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n","  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n","  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n","  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n","  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n","  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n","  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n","  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n","  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n","  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n","  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n","  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n","  inflating: FastQC/README.md        \n","  inflating: FastQC/README.txt       \n","  inflating: FastQC/RELEASE_NOTES.txt  \n","  inflating: FastQC/run_fastqc.bat   \n","  inflating: FastQC/sam-1.103.jar    \n","   creating: FastQC/Templates/\n","  inflating: FastQC/Templates/fastqc2fo.xsl  \n","  inflating: FastQC/Templates/header_template.html  \n","   creating: FastQC/Templates/Icons/\n"," extracting: FastQC/Templates/Icons/error.png  \n"," extracting: FastQC/Templates/Icons/fastqc_icon.png  \n"," extracting: FastQC/Templates/Icons/tick.png  \n"," extracting: FastQC/Templates/Icons/warning.png  \n","   creating: FastQC/uk/\n","   creating: FastQC/uk/ac/\n","   creating: FastQC/uk/ac/babraham/\n","   creating: FastQC/uk/ac/babraham/FastQC/\n","   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n","  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n","   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n","  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n","   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n","  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n","   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n","  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n","   creating: FastQC/uk/ac/babraham/FastQC/Help/\n","  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n","   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n","   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n","   creating: FastQC/uk/ac/babraham/FastQC/Report/\n","  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n","   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n"," extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n"," extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n"," extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n"," extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n"," extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n","   creating: FastQC/uk/ac/babraham/FastQC/Results/\n","  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n","   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n","  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n","   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n","  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n","   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n","  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n","   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n","  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n","   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n","  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n","  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n","Started analysis of Final_YoungSARScov2.fastq\n","Approx 5% complete for Final_YoungSARScov2.fastq\n","Approx 10% complete for Final_YoungSARScov2.fastq\n","Approx 15% complete for Final_YoungSARScov2.fastq\n","Approx 20% complete for Final_YoungSARScov2.fastq\n","Approx 25% complete for Final_YoungSARScov2.fastq\n","Approx 30% complete for Final_YoungSARScov2.fastq\n","Approx 35% complete for Final_YoungSARScov2.fastq\n","Approx 40% complete for Final_YoungSARScov2.fastq\n","Approx 45% complete for Final_YoungSARScov2.fastq\n","Approx 50% complete for Final_YoungSARScov2.fastq\n","Approx 55% complete for Final_YoungSARScov2.fastq\n","Approx 60% complete for Final_YoungSARScov2.fastq\n","Approx 65% complete for Final_YoungSARScov2.fastq\n","Approx 70% complete for Final_YoungSARScov2.fastq\n","Approx 75% complete for Final_YoungSARScov2.fastq\n","Approx 80% complete for Final_YoungSARScov2.fastq\n","Approx 85% complete for Final_YoungSARScov2.fastq\n","Approx 90% complete for Final_YoungSARScov2.fastq\n","Approx 95% complete for Final_YoungSARScov2.fastq\n","Approx 100% complete for Final_YoungSARScov2.fastq\n","Analysis complete for Final_YoungSARScov2.fastq\n"]}]},{"cell_type":"code","source":["# RNA sample alignments to genome\n","!apt-get install hisat2"],"metadata":{"id":"3waZYR0xLtVN"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["!hisat2 -x /content/drive/MyDrive/SARSdata/mm10/genome -p 66 -U /content/drive/MyDrive/SARSdata/Final_MidagePBS.fastq -S /content/drive/MyDrive/SARSdata/Aln_Midage_PBS.sam"],"metadata":{"id":"-tJvyZ55OYrx"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["!hisat2 -x /content/drive/MyDrive/SARSdata/mm10/genome -p 66 -U /content/drive/MyDrive/SARSdata/Final_MidageSARScov2.fastq -S /content/drive/MyDrive/SARSdata/Aln_Midage_SARScov2.sam"],"metadata":{"id":"scmkUBwCPjmR"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["!hisat2 -x /content/drive/MyDrive/SARSdata/mm10/genome -p 66 -U /content/drive/MyDrive/SARSdata/Final_YoungSARScov2.fastq -S /content/drive/MyDrive/SARSdata/Aln_Young_SARScov2.sam"],"metadata":{"id":"xQPadLh3cxCA"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["!hisat2 -x /content/drive/MyDrive/SARSdata/mm10/genome -p 66 -U /content/drive/MyDrive/SARSdata/Final_YoungPBS.fastq -S /content/drive/MyDrive/SARSdata/Aln_Young_PBS.sam"],"metadata":{"id":"235PUQMFc8fQ"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["#subread: a general-purpose read aligner which can align both genomic DNA-seq and RNA-seq reads.\n","#It can also be used to discover genomic mutations including short indels and structural variants.\n","#documentation: https://subread.sourceforge.net/\n","!apt-get install subread"],"metadata":{"id":"fgzMeoA740no"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# featureCounts - read summarization program that counts mapped reads for genomic features\n","# input: SAM/BAM files + an annotation file (including chromosomal coordinates of features)\n","# output: #s of reads assigned to features (or meta-features)\n","!featureCounts -a /content/Mus_musculus.GRCm38.102.gtf -o counts.txt /content/drive/MyDrive/SARSdata/Aln_Midage_PBS.sam"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"TMNI2Z3s6FVP","executionInfo":{"status":"ok","timestamp":1712804060468,"user_tz":420,"elapsed":28784,"user":{"displayName":"Emily White","userId":"02727269720942967183"}},"outputId":"06910b18-b90a-4311-c15b-7440cfbdf9f9"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["\n","       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n","       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n","       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n","       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n","       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n","       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n","\t  v2.0.3\n","\n","//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n","||  \u001b[0m                                                                          ||\n","||             Input files : \u001b[36m1 SAM file  \u001b[0m \u001b[0m                                    ||\n","||  \u001b[0m                                                                          ||\n","||                           \u001b[36mAln_Midage_PBS.sam\u001b[0m \u001b[0m                              ||\n","||  \u001b[0m                                                                          ||\n","||             Output file : \u001b[36mcounts.txt\u001b[0m \u001b[0m                                      ||\n","||                 Summary : \u001b[36mcounts.txt.summary\u001b[0m \u001b[0m                              ||\n","||              Paired-end : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n","||        Count read pairs : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n","||              Annotation : \u001b[36mMus_musculus.GRCm38.102.gtf (GTF)\u001b[0m \u001b[0m               ||\n","||      Dir for temp files : \u001b[36m./\u001b[0m \u001b[0m                                              ||\n","||  \u001b[0m                                                                          ||\n","||                 Threads : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n","||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n","||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n","|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n","||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n","||  \u001b[0m                                                                          ||\n","\\\\============================================================================//\n","\n","//=================================\u001b[36m Running \u001b[0m==================================\\\\\n","||  \u001b[0m                                                                          ||\n","|| Load annotation file Mus_musculus.GRCm38.102.gtf \u001b[0m... \u001b[0m                      ||\n","||    Features : \u001b[36m843712\u001b[0m \u001b[0m                                                      ||\n","||    Meta-features : \u001b[36m55487\u001b[0m \u001b[0m                                                  ||\n","||    Chromosomes/contigs : \u001b[36m45\u001b[0m \u001b[0m                                               ||\n","||  \u001b[0m                                                                          ||\n","|| Process SAM file Aln_Midage_PBS.sam... \u001b[0m                                    ||\n","||    Single-end reads are included. \u001b[0m                                         ||\n","||    Total alignments : \u001b[36m5707686\u001b[0m \u001b[0m                                             ||\n","||    Successfully assigned alignments : \u001b[36m3472963 (60.8%)\u001b[0m \u001b[0m                     ||\n","||    Running time : \u001b[36m0.38 minutes\u001b[0m \u001b[0m                                            ||\n","||  \u001b[0m                                                                          ||\n","|| Write the final count table. \u001b[0m                                              ||\n","|| Write the read assignment summary. \u001b[0m                                        ||\n","||  \u001b[0m                                                                          ||\n","|| Summary of counting results can be found in file \"counts.txt.summary\" \u001b[0m     ||\n","||  \u001b[0m                                                                          ||\n","\\\\============================================================================//\n","\n"]}]},{"cell_type":"code","source":["!mv counts.txt MidagePBS.txt\n","!cp MidagePBS.txt /content/drive/MyDrive/SARSdata/\n","!mv counts.txt.summary MidagePBScounts.txt.summary\n","!cp MidagePBScounts.txt.summary /content/drive/MyDrive/SARSdata/"],"metadata":{"id":"YFUceoKmEry_"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["!featureCounts -a /content/Mus_musculus.GRCm38.102.gtf -o counts.txt /content/drive/MyDrive/SARSdata/Aln_Midage_SARScov2.sam"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"5yij1aefEgNM","executionInfo":{"status":"ok","timestamp":1712804615708,"user_tz":420,"elapsed":33876,"user":{"displayName":"Emily White","userId":"02727269720942967183"}},"outputId":"79051817-d362-4657-8498-ff974c2ef7e2"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["\n","       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n","       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n","       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n","       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n","       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n","       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n","\t  v2.0.3\n","\n","//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n","||  \u001b[0m                                                                          ||\n","||             Input files : \u001b[36m1 SAM file  \u001b[0m \u001b[0m                                    ||\n","||  \u001b[0m                                                                          ||\n","||                           \u001b[36mAln_Midage_SARScov2.sam\u001b[0m \u001b[0m                         ||\n","||  \u001b[0m                                                                          ||\n","||             Output file : \u001b[36mcounts.txt\u001b[0m \u001b[0m                                      ||\n","||                 Summary : \u001b[36mcounts.txt.summary\u001b[0m \u001b[0m                              ||\n","||              Paired-end : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n","||        Count read pairs : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n","||              Annotation : \u001b[36mMus_musculus.GRCm38.102.gtf (GTF)\u001b[0m \u001b[0m               ||\n","||      Dir for temp files : \u001b[36m./\u001b[0m \u001b[0m                                              ||\n","||  \u001b[0m                                                                          ||\n","||                 Threads : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n","||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n","||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n","|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n","||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n","||  \u001b[0m                                                                          ||\n","\\\\============================================================================//\n","\n","//=================================\u001b[36m Running \u001b[0m==================================\\\\\n","||  \u001b[0m                                                                          ||\n","|| Load annotation file Mus_musculus.GRCm38.102.gtf \u001b[0m... \u001b[0m                      ||\n","||    Features : \u001b[36m843712\u001b[0m \u001b[0m                                                      ||\n","||    Meta-features : \u001b[36m55487\u001b[0m \u001b[0m                                                  ||\n","||    Chromosomes/contigs : \u001b[36m45\u001b[0m \u001b[0m                                               ||\n","||  \u001b[0m                                                                          ||\n","|| Process SAM file Aln_Midage_SARScov2.sam... \u001b[0m                               ||\n","||    Single-end reads are included. \u001b[0m                                         ||\n","||    Total alignments : \u001b[36m5849472\u001b[0m \u001b[0m                                             ||\n","||    Successfully assigned alignments : \u001b[36m2889609 (49.4%)\u001b[0m \u001b[0m                     ||\n","||    Running time : \u001b[36m0.42 minutes\u001b[0m \u001b[0m                                            ||\n","||  \u001b[0m                                                                          ||\n","|| Write the final count table. \u001b[0m                                              ||\n","|| Write the read assignment summary. \u001b[0m                                        ||\n","||  \u001b[0m                                                                          ||\n","|| Summary of counting results can be found in file \"counts.txt.summary\" \u001b[0m     ||\n","||  \u001b[0m                                                                          ||\n","\\\\============================================================================//\n","\n"]}]},{"cell_type":"code","source":["!mv counts.txt MidageSARScounts.txt\n","!cp MidageSARScounts.txt /content/drive/MyDrive/SARSdata/\n","!mv counts.txt.summary MidageSARScounts.txt.summary\n","!cp MidageSARScounts.txt.summary /content/drive/MyDrive/SARSdata/"],"metadata":{"id":"aHbuFUujGwPL"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["!featureCounts -a /content/Mus_musculus.GRCm38.102.gtf -o counts.txt /content/drive/MyDrive/SARSdata/Aln_Young_SARScov2.sam"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"wyNeOp0SH2ml","executionInfo":{"status":"ok","timestamp":1712804888414,"user_tz":420,"elapsed":34866,"user":{"displayName":"Emily White","userId":"02727269720942967183"}},"outputId":"34f7100e-96b6-4bb3-aa86-c691c01fec01"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["\n","       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n","       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n","       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n","       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n","       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n","       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n","\t  v2.0.3\n","\n","//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n","||  \u001b[0m                                                                          ||\n","||             Input files : \u001b[36m1 SAM file  \u001b[0m \u001b[0m                                    ||\n","||  \u001b[0m                                                                          ||\n","||                           \u001b[36mAln_Young_SARScov2.sam\u001b[0m \u001b[0m                          ||\n","||  \u001b[0m                                                                          ||\n","||             Output file : \u001b[36mcounts.txt\u001b[0m \u001b[0m                                      ||\n","||                 Summary : \u001b[36mcounts.txt.summary\u001b[0m \u001b[0m                              ||\n","||              Paired-end : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n","||        Count read pairs : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n","||              Annotation : \u001b[36mMus_musculus.GRCm38.102.gtf (GTF)\u001b[0m \u001b[0m               ||\n","||      Dir for temp files : \u001b[36m./\u001b[0m \u001b[0m                                              ||\n","||  \u001b[0m                                                                          ||\n","||                 Threads : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n","||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n","||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n","|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n","||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n","||  \u001b[0m                                                                          ||\n","\\\\============================================================================//\n","\n","//=================================\u001b[36m Running \u001b[0m==================================\\\\\n","||  \u001b[0m                                                                          ||\n","|| Load annotation file Mus_musculus.GRCm38.102.gtf \u001b[0m... \u001b[0m                      ||\n","||    Features : \u001b[36m843712\u001b[0m \u001b[0m                                                      ||\n","||    Meta-features : \u001b[36m55487\u001b[0m \u001b[0m                                                  ||\n","||    Chromosomes/contigs : \u001b[36m45\u001b[0m \u001b[0m                                               ||\n","||  \u001b[0m                                                                          ||\n","|| Process SAM file Aln_Young_SARScov2.sam... \u001b[0m                                ||\n","||    Single-end reads are included. \u001b[0m                                         ||\n","||    Total alignments : \u001b[36m5753032\u001b[0m \u001b[0m                                             ||\n","||    Successfully assigned alignments : \u001b[36m3077535 (53.5%)\u001b[0m \u001b[0m                     ||\n","||    Running time : \u001b[36m0.44 minutes\u001b[0m \u001b[0m                                            ||\n","||  \u001b[0m                                                                          ||\n","|| Write the final count table. \u001b[0m                                              ||\n","|| Write the read assignment summary. \u001b[0m                                        ||\n","||  \u001b[0m                                                                          ||\n","|| Summary of counting results can be found in file \"counts.txt.summary\" \u001b[0m     ||\n","||  \u001b[0m                                                                          ||\n","\\\\============================================================================//\n","\n"]}]},{"cell_type":"code","source":["!mv counts.txt YoungSARScounts.txt\n","!cp YoungSARScounts.txt /content/drive/MyDrive/SARSdata/\n","!mv counts.txt.summary YoungSARScounts.txt.summary\n","!cp YoungSARScounts.txt.summary /content/drive/MyDrive/SARSdata/"],"metadata":{"id":"VF1tUU4zHjz7"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["!featureCounts -a /content/Mus_musculus.GRCm38.102.gtf -o counts.txt /content/drive/MyDrive/SARSdata/Aln_Young_PBS.sam"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"LrCMdY6aH5LK","executionInfo":{"status":"ok","timestamp":1712805195432,"user_tz":420,"elapsed":23621,"user":{"displayName":"Emily White","userId":"02727269720942967183"}},"outputId":"a5d414e5-d312-4017-8d44-eaaac14a3e2d"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["\n","       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n","       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n","       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n","       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n","       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n","       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n","\t  v2.0.3\n","\n","//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n","||  \u001b[0m                                                                          ||\n","||             Input files : \u001b[36m1 SAM file  \u001b[0m \u001b[0m                                    ||\n","||  \u001b[0m                                                                          ||\n","||                           \u001b[36mAln_Young_PBS.sam\u001b[0m \u001b[0m                               ||\n","||  \u001b[0m                                                                          ||\n","||             Output file : \u001b[36mcounts.txt\u001b[0m \u001b[0m                                      ||\n","||                 Summary : \u001b[36mcounts.txt.summary\u001b[0m \u001b[0m                              ||\n","||              Paired-end : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n","||        Count read pairs : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n","||              Annotation : \u001b[36mMus_musculus.GRCm38.102.gtf (GTF)\u001b[0m \u001b[0m               ||\n","||      Dir for temp files : \u001b[36m./\u001b[0m \u001b[0m                                              ||\n","||  \u001b[0m                                                                          ||\n","||                 Threads : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n","||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n","||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n","|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n","||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n","||  \u001b[0m                                                                          ||\n","\\\\============================================================================//\n","\n","//=================================\u001b[36m Running \u001b[0m==================================\\\\\n","||  \u001b[0m                                                                          ||\n","|| Load annotation file Mus_musculus.GRCm38.102.gtf \u001b[0m... \u001b[0m                      ||\n","||    Features : \u001b[36m843712\u001b[0m \u001b[0m                                                      ||\n","||    Meta-features : \u001b[36m55487\u001b[0m \u001b[0m                                                  ||\n","||    Chromosomes/contigs : \u001b[36m45\u001b[0m \u001b[0m                                               ||\n","||  \u001b[0m                                                                          ||\n","|| Process SAM file Aln_Young_PBS.sam... \u001b[0m                                     ||\n","||    Single-end reads are included. \u001b[0m                                         ||\n","||    Total alignments : \u001b[36m5734806\u001b[0m \u001b[0m                                             ||\n","||    Successfully assigned alignments : \u001b[36m3066246 (53.5%)\u001b[0m \u001b[0m                     ||\n","||    Running time : \u001b[36m0.22 minutes\u001b[0m \u001b[0m                                            ||\n","||  \u001b[0m                                                                          ||\n","|| Write the final count table. \u001b[0m                                              ||\n","|| Write the read assignment summary. \u001b[0m                                        ||\n","||  \u001b[0m                                                                          ||\n","|| Summary of counting results can be found in file \"counts.txt.summary\" \u001b[0m     ||\n","||  \u001b[0m                                                                          ||\n","\\\\============================================================================//\n","\n"]}]},{"cell_type":"code","source":["# !mv counts.txt YoungPBScounts.txt\n","# !cp YoungPBScounts.txt /content/drive/MyDrive/SARSdata/\n","!mv counts.txt.summary YoungPBScounts.txt.summary\n","!cp YoungPBScounts.txt.summary /content/drive/MyDrive/SARSdata/"],"metadata":{"id":"c3HOZ82gHl-0"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["from google.colab import files\n","# files.download('MidagePBScounts.txt')\n","files.download('MidageSARScounts.txt')\n","files.download('YoungSARScounts.txt')\n","files.download('YoungPBScounts.txt')"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":17},"id":"TQJLpiCC-jB2","executionInfo":{"status":"ok","timestamp":1712805343716,"user_tz":420,"elapsed":227,"user":{"displayName":"Emily White","userId":"02727269720942967183"}},"outputId":"b6065664-f310-41cd-e993-a7fb908abbc1"},"execution_count":null,"outputs":[{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["\n","    async function download(id, filename, size) {\n","      if (!google.colab.kernel.accessAllowed) {\n","        return;\n","      }\n","      const div = document.createElement('div');\n","      const label = document.createElement('label');\n","      label.textContent = `Downloading \"${filename}\": `;\n","      div.appendChild(label);\n","      const progress = document.createElement('progress');\n","      progress.max = size;\n","      div.appendChild(progress);\n","      document.body.appendChild(div);\n","\n","      const buffers = [];\n","      let downloaded = 0;\n","\n","      const channel = await google.colab.kernel.comms.open(id);\n","      // Send a message to notify the kernel that we're ready.\n","      channel.send({})\n","\n","      for await (const message of channel.messages) {\n","        // Send a message to notify the kernel that we're ready.\n","        channel.send({})\n","        if (message.buffers) {\n","          for (const buffer of message.buffers) {\n","            buffers.push(buffer);\n","            downloaded += buffer.byteLength;\n","            progress.value = downloaded;\n","          }\n","        }\n","      }\n","      const blob = new Blob(buffers, {type: 'application/binary'});\n","      const a = document.createElement('a');\n","      a.href = window.URL.createObjectURL(blob);\n","      a.download = filename;\n","      div.appendChild(a);\n","      a.click();\n","      div.remove();\n","    }\n","  "]},"metadata":{}},{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["download(\"download_a9b8e69c-1f46-42e2-a9cb-fa96d45a6ba5\", \"MidageSARScounts.txt\", 20770164)"]},"metadata":{}},{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["\n","    async function download(id, filename, size) {\n","      if (!google.colab.kernel.accessAllowed) {\n","        return;\n","      }\n","      const div = document.createElement('div');\n","      const label = document.createElement('label');\n","      label.textContent = `Downloading \"${filename}\": `;\n","      div.appendChild(label);\n","      const progress = document.createElement('progress');\n","      progress.max = size;\n","      div.appendChild(progress);\n","      document.body.appendChild(div);\n","\n","      const buffers = [];\n","      let downloaded = 0;\n","\n","      const channel = await google.colab.kernel.comms.open(id);\n","      // Send a message to notify the kernel that we're ready.\n","      channel.send({})\n","\n","      for await (const message of channel.messages) {\n","        // Send a message to notify the kernel that we're ready.\n","        channel.send({})\n","        if (message.buffers) {\n","          for (const buffer of message.buffers) {\n","            buffers.push(buffer);\n","            downloaded += buffer.byteLength;\n","            progress.value = downloaded;\n","          }\n","        }\n","      }\n","      const blob = new Blob(buffers, {type: 'application/binary'});\n","      const a = document.createElement('a');\n","      a.href = window.URL.createObjectURL(blob);\n","      a.download = filename;\n","      div.appendChild(a);\n","      a.click();\n","      div.remove();\n","    }\n","  "]},"metadata":{}},{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["download(\"download_70790351-25cd-4a10-82be-38ab12cd2b13\", \"YoungSARScounts.txt\", 20770677)"]},"metadata":{}},{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["\n","    async function download(id, filename, size) {\n","      if (!google.colab.kernel.accessAllowed) {\n","        return;\n","      }\n","      const div = document.createElement('div');\n","      const label = document.createElement('label');\n","      label.textContent = `Downloading \"${filename}\": `;\n","      div.appendChild(label);\n","      const progress = document.createElement('progress');\n","      progress.max = size;\n","      div.appendChild(progress);\n","      document.body.appendChild(div);\n","\n","      const buffers = [];\n","      let downloaded = 0;\n","\n","      const channel = await google.colab.kernel.comms.open(id);\n","      // Send a message to notify the kernel that we're ready.\n","      channel.send({})\n","\n","      for await (const message of channel.messages) {\n","        // Send a message to notify the kernel that we're ready.\n","        channel.send({})\n","        if (message.buffers) {\n","          for (const buffer of message.buffers) {\n","            buffers.push(buffer);\n","            downloaded += buffer.byteLength;\n","            progress.value = downloaded;\n","          }\n","        }\n","      }\n","      const blob = new Blob(buffers, {type: 'application/binary'});\n","      const a = document.createElement('a');\n","      a.href = window.URL.createObjectURL(blob);\n","      a.download = filename;\n","      div.appendChild(a);\n","      a.click();\n","      div.remove();\n","    }\n","  "]},"metadata":{}},{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.Javascript object>"],"application/javascript":["download(\"download_aeac5e60-5735-490c-9f6d-fd6a97be556f\", \"YoungPBScounts.txt\", 20771234)"]},"metadata":{}}]}],"metadata":{"colab":{"provenance":[{"file_id":"1BfFcFUeW5TM7h7c0QLQumqFNUTD3fD7V","timestamp":1712169077425}]},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}