---
title: "Bioinformatics in R WGCNA"
author: "J. Cesar Ignacio Espinoza - Cesar   "
date: "Week 05: April 15th and 17th 2024"
output: 
  html_document: 
    highlight: espresso
    theme: cerulean
---

### This class will incorporate a bit of ML.

We will be performing a WGNCA, before proceeding test yourself and make sure you understand what weighted. Gene_network and correlation mean?


## The dataset.  
 we will be working with the dataset "	Systems biological assessment of immunity to severe and mild COVID-19 infections" 
 
RNAseq analysis of PBMCs in a group of 17 COVID-19 subjects and 17 healthy controls


```{r setup}
    ### Edit this line so your notebook renders properly
    knitr::opts_knit$set(root.dir = normalizePath("C:/Users/13035/OneDrive/Desktop/Bioinformatics_R/WGNCA")) 
```
We will be using the package called WGCNA, if you do not have it install, please run this cell, once it is installed comment it!
```{r}
#if (!require("BiocManager", quietly = TRUE))
 #   install.packages("BiocManager")
#BiocManager::install("WGCNA")
```

We now load the libraries
```{r message=FALSE, warning=FALSE, paged.print=FALSE, results="hide"}
# We first need to import the important libnrary for today's class, dplyr
library(WGCNA)
library(dplyr)
library(readr)
library(DESeq2)
library(ggplot2)
```

Load the data (Counts table and metadata from canvas site)

```{r}
### Run this chunk to import the counts table and metadata into your evironment.
counts <- read.csv('GSE152418RawCounts.csv', header = TRUE, row.names = 1)
metadata <- read.csv('GSE152418Metadata.csv', header = TRUE, row.names = 1)
# meta data could include things like seq. technology, treatment types, age/sex/location per sample
```

### QC:
Here we wanna explore to see if the dataset that we have is good for analysis
We are going to use a function called goodSamplesGenes(). Use the cell below to displat the help page of this function, figure out if you can run it

```{r}
#?goodSamplesGenes()
#? tells you about the function
goodSamplesGenes(counts)
# quickly tells you if sample/genes are good quality 
# datExpr	expression data. A matrix or data frame in which columns are genes and rows ar samples.
# Flip columns with rows --> transpose
####goodSamplesGenes(t(Counts))
gsG <- goodSamplesGenes(t(counts))
## type gSG in console --> FALSE values are flagged
```

Subset your data so only the genes that passed the filter are kept

```{r}
good_counts <- counts[gsG$goodGenes,] #, is to look at these rows
```

Another way to detect outliers is to perform hierarchical clustering of all the samples. If you do that you should be able to see if some data points are too far from the rest of the samples.

```{r}

temptree <- hclust(dist(t(good_counts)), method = 'complete')
plot(temptree)
#remove outliers (first and last genes)
#distance, expects t = transpose 
```

perform hclustering on the data, **HINT!!!** Double chekc tnat columns and rows are as the program expects them!

```{r}
### Write your code here

```

Outliers are literally that samples taht are far from each other, we can also look at that by applying dimensionality reduction, one of the most common techniques is PCA. run the cell below to go to the help page for PCA

```{r}
pca <- prcomp(t(good_counts))
```


```{r}

pca$x

```
```{r}

ggplot(data = pca$x, aes(x=PC1, y=PC2)) + geom_point() +  geom_text(label=rownames(pca$x))

```
# Filter the data to remove bad samples
**HINT** Use DPlyr
```{r}

really_good_counts <- good_counts %>%
  dplyr::select(-GSM4614985) %>%
  dplyr::select(-GSM4614993) %>%
  dplyr::select(-GSM4615000)
```

#Normalization. 

The 'easiest' way will be to run DESEq2 and use the normalized counts object from DESeq2, Look at your past notes and run it below. You have all you need but you might need to play with the metadata file.
HINT : df[!(row.names(df) %in% row_names_df_to_remove),] ### 

```{r}
### WRITE YOUR CODE HERE, ALSO RENAME THE COLUMNS OF METADATA SO IT IS EASIER TO READ, REMOVE 'DOTS' 
  ### manipulate the data frame to remove 
remove_list <- c('GSM4614985', 'GSM4614993', 'GSM4615000')

#renaming something
phenotype <- metadata[!rownames(metadata) %in% c('GSM4614985', 'GSM4614993', 'GSM4615000'),]
temp_rename <-rename  # as a place holder
rename <- dplyr::rename #because the rename function has duplicate applications
phenotype
```
```{r}
new_pheno <- phenotype %>%
  rename(disease_state = disease.state.ch1) %>%
  rename(gender = gender.ch1) %>%
  rename(severity = severity.ch1) %>%
  rename(days_onset = days_post_symptom_onset.ch1)%>%
  dplyr::select(-geographical.location.ch1)
```
```{r}
new_pheno
```

```{r}
library(DESeq2)
dds <- DESeqDataSetFromMatrix(countData = really_good_counts, 
                              colData = new_pheno,
                              design = ~ 1) # all vs. all
```

 Now remove the genes with counts < 15 in more than 75% of samples (31*0.75 ~ 23)
This number is coming from the WGCNA recommendations

```{r}
dds75 <- dds[rowSums(counts(dds)) >= 23]
``` 

```{r}
dds_norm <- vst(dds75) #vst takes the harmonic mean vs. the arimethic? mean
norm_gene_exp <- t(assay(dds_norm))

```
#We can finally start with our WGNCA data analysis

First we pick up a soft threshold modify the power vector below 

```{r}
## correlation method --> threshold --> adjacency matrix
sft <- pickSoftThreshold(norm_gene_exp, 
                  powerVector = c(1:20),
                  networkType = "signed", 
                  verbose = 2) #how much of the output you want to see
```
```{r}
deseq_ob <- DESeq(dds75)
res <- results(deseq_ob, alpha = 0.05)
View(res)
res_filtered <- res[row.names(res) %in% as.vector(row.names(diff.df)),]
res_filtered
```

```{r}
rf.df <- as.data.frame(res_filtered)
rf.df <- rf.df %>%
  filter(padj < 0.05, log2FoldChange > 2, baseMean > 100)
genes_to_plot <- rownames(rf.df)
View(genes_to_plot)
```

You can acess the results with sft$fitIndices. We are going to pick a power that gives us the higherst R2 and the lowest mean K. 

**HINT plot the data!** First plot Power vs r2
```{r}
ggplot(data = sft$fitIndices, aes(x = Power, y= SFT.R.sq)) +
  geom_point()
  
```

Then Plot Power vs mean.k
```{r}
ggplot(data = sft$fitIndices, aes(x = Power, y= mean.k.)) +
  geom_point()
```
After you pick up a threshold we are ready to run our data analysis

```{r}
temp_cor <-  cor
cor <- WGCNA::cor
norm_gene_exp[] <- sapply(norm_gene_exp, as.numeric)

bwm <- blockwiseModules(norm_gene_exp, 
                 maxBlockSize = 5,000, # max start of genes
                 TOMType = "signed", # corelation: sees if something is up the other down; negative correlation (moving in the same direction)
                 power = 10, # #of clusters?
                 mergeCutHeight = 0.2, #this is the distance threshold line thing; farthest apart is 1
                 numericLabels = FALSE, 
                 randomSeed = 1234, 
                 verbose = 2)
```

#explore the bwm object, 
how many modules are there? 
What us the largest module?
What is the smallest?
```{r}
table(bwm$colors) # color assignment of each gene --> how many of clusters of genes moving together in the same direction across different treatments 
```

```{r}
## RUN THIS AS IS, IT WILL PLOT THE COLORS AND DENDROGRAM
###  can review the manual for WGCNA
### just a display; all the data that we need is in the object
## each tree is a gene; the reverse peaks are moving together
##### circos plot?
## gene names --> plot gene richment analysis 

mergedColors = labels2colors(bwm$colors)
plotDendroAndColors(
  bwm$dendrograms[[1]],
  mergedColors[bwm$blockGenes[[1]]],
  "Module colors",
  dendroLabels = FALSE,
  hang = 0.03,
  addGuide = TRUE,
  guideHang = 0.05 )
```
# Now we can correlate our findings with phenotypic states of patients

```{r}

## hub1encoded?
#### 

traits <- new_pheno_name %>%
  mutate(disease_state_bin = ifelse(grepl('COVID', disease_state),1,0)) # if it hs the word covid -> assign 1 , not assign 0; mutate is to add a new column to  the dateframe 

traits   # adds a new column/variable: disease state (1,0); convterts disease state column into binary 

# how would you code severity--? more than one column Moderate yes or no, ICU yes or no
traits <- new_pheno_name %>%
  mutate(ICU_bin = ifelse(grepl('ICU', severity),1,0)) %>%
  mutate(Severe_bin = ifelse(grepl('Severe', severity),1,0)) %>%
  mutate(Moderate_bin = ifelse(grepl('Moderate', severity),1,0)) %>%
  mutate(Con_bin = ifelse(grepl('Convalescent', severity),1,0)) 

traits

```

```{r}
# what if we want to select only the colums we added
new_pheno
# correlation the modules that we just identified with the traits to see if any of the gene clusters are moving with traits
# highest correlation value --> which gene networks associated to disease state!!!! (correlations is an object)
# calculate the correlations between modules of co-expressed genes (MEs) and traits in a dataset called "bwn"
###correlations = cor(bwn$MEs, traits, use = 'p')

pvalues = corPvalueStudent(correlations, 31)
# tells gene with highest connectivity in those modules 

chooseTopHubInEachModule(
  norm_gene_exp, # datExpr
  mergedColors, #colorh,
  omitColors = "salmon",
  power = 15
  type = "singed"
)

library(ComplexHeatmap)
Heatmap(Corrleations)
```
### k was to find few genes with high degree (hub proteins)
```{r}
file_path <-"C:/Users/13035/OneDrive/Desktop/Bioinformatics_R/HW1/res.csv"
write.csv(res, file = file_path , row.names = TRUE)
```


